default.nix

```diff
@@ -150,14 +150,10 @@ let iosSupport = system == "x86_64-darwin";
         outPath = filterGit p;
       };
 
-    # All imports of sources need to go here, so that they can be explicitly cached
-    sources = {
-      ghcjs8_0 = {
-        boot = hackGet ./ghcjs-8.0/boot;
-        shims = hackGet ./ghcjs-8.0/shims;
-        ghcjs = hackGet ./ghcjs-8.0/ghcjs;
-      };
-    };
+    # Make an attribute set of source derivations for a directory containing thunks:
+    thunkSet = dir: lib.mapAttrs (name: _: hackGet (dir + "/${name}")) (builtins.readDir dir);
```

Comment:
Things that can be `hackGet`ed aren't necessarily "thunk"s in the `ob thunk` parlance. The target *must* have `default.nix`. So I'm not sure thunk is the best name here, unless it's actually true?

Reply:
Hmm, to me the name "thunk" suggests the expandable/collapsible source and not inherently that it contains a `default.nix`. `ob thunk` does happen to always put in a `default.nix` even if the repository contains none so that you can seamlessly add one later, but that seems non-essential to me.

I'm happy to change the name though if we have a better term for just what `hackGet` expects.

Reply:
I agree that the term "thunk" is attractive here. However, `ob thunk` doesn't actually *add* anything. It actually *requires* that the upstream repository have a `default.nix` or it won't work. Sometimes/often you can't get a `default.nix` into the upstream repo...which is why `hackGet` is used instead. Point being, there is a clear semantic difference between these two things and mingling the terms is just going to get really confusing. On another note, I don't like that `ob thunk` doesn't support source repos and `hackGet` isn't the best name ever...I'd be much happier merging the two things somehow.

Reply:
This is a very fine distinction, and one that I've only come to see by being bitten by it. Obelisk thunks are designed to allow you to use `nix` to pull in a dependency *regardless of its evaluation state* (i.e. packed or unpacked). If a target repo doesn't have a `default.nix` then any nix referring to the `default.nix` will not work because the `default.nix` in a packed thunk merely imports the `default.nix` from the repo. And if you unpack the thunk, there is *no* `default.nix` to import. So your nix is now broken. And this is the behavior that `hackGet` doesn't have, for a reason.

Reply:
I have thought about this in the past, and my *preference* would be to merge the two ideas and make it such that *all* thunks expose an extra API that lets them work either as "imported thunks" or as "raw source thunks".

Reply:
There was some discussion it might be better to go the other way and *never* assume that obelisk thunks when unpacked can be imported with `default.nix`. Beyond just separating concerns (src obtainment vs build systems), libraries ideally wouldn't need any nix (except for their own CI) because `callCabal2nix` works just fine.